### ISSUE TITLE
Building a much smaller binary with SBCL core compression, portably with ASDF

### ISSUE THREAD

**Issue by vindarel** [lisp-tips/lisp-tips#20]
In the [Next](https://next.atlas.engineer/) browser, building with SBCL's core compression gave us a 23MB executable, instead of 120MB, for a loss of a dozen milliseconds of start uptime, which is still under the 50ms, so not noticeable.

Your SBCL must be built with core compression, see the documentation: http://www.sbcl.org/manual/#Saving-a-Core-Image

Is it the case ?

```
(find :sb-core-compression *features*)
:SB-CORE-COMPRESSION
```

Yes, it is the case with this SBCL 1.4.5.debian.

WIth pure SBCL, we give an argument to `save-lisp-and-die`, where `:compression` 

> may also be an integer from -1 to 9, corresponding to zlib compression levels, or t (which is equivalent to the default compression level, -1). 

However, how to do this portably with ASDF ? Add this in your .asd:

```
#+sb-core-compression
(defmethod asdf:perform ((o asdf:image-op) (c asdf:system))
  (uiop:dump-image (asdf:output-file o c) :executable t :compression t))
```

For users having an out of date or poorly built SBCL, they can use the [Guix](https://guix.gnu.org/) package manager. Also, certainly [Roswell](https://github.com/roswell/roswell/).

(thanks to @ambrevar, from https://github.com/LispCookbook/cl-cookbook/issues/262)

**Comment by svetlyak40wt** [comment]
Also, [Deploy](http://quickdocs.org/deploy/) could be used to build an image. It will use compression if it available.

And Deploy also collects all `*.so` files of dependencies, like libssl.co in the `bin` subdirectory.

**Comment by Ambrevar** [comment]
For that I would recommend [Guix pack](http://guix.gnu.org/blog/2018/tarballs-the-ultimate-container-image-format/) over deploy, since it
works on any file system (e.g. it won't fail if the host system has
/lib64 instead of /lib, or vice versa).

**Comment by svetlyak40wt** [comment]
@Ambrevar Guix is a tool for distribution. Deploy is a helper for building executables.

By the way, can I use Guix on OSX?

**Comment by Ambrevar** [comment]
In this context I believe you were talking about bundling .so for
distribution.  Or did you mean something else?

Guix on macOS is still not a thing :/

**Comment by svetlyak40wt** [comment]
No, Deploy just gathers all dependency *.so files into one subdirectory. It doesn't package them in any other way.

**Comment by Ambrevar** [comment]
Oh OK, my bad.
But I'm curious: what for?

**Comment by svetlyak40wt** [comment]
For further bundling and distribution.
